Song Creation form is not sufficiently user friendly #305

Pyther99 opened the pull request:
Depends on #817.
Closes #305.
Closes #14.
Partially implements #368. (Romanization link)

Implementation of @dset0x's patch for the song creation page. 

- [x] This PR needs to be merged after #817 to include the new alert
- [x]  The current alert texts need to be split 
- [x] The new alert texts need to be translated

Needs to be tested on the beta site.

Pyther99 commented on VocaDbWeb/wwwroot/Content/Styles/base.less:
```diff
@@ -74,23 +74,23 @@ body {
-	}
+		padding-right: 10px;
+		padding-left: 10px;
+	}
 }
 
-.row-fluid .rightFrame {
+.row-fluid .rightFrame {
 	@media (min-width: @left-menu-breakpoint) {
-		margin-left: 1%;
+		margin-left: 1%;
 		width: 84%;
-	}
+	}
 	@media (max-width: @left-menu-breakpoint) {
 		margin-left: 0;
 		margin-right: 0;
 		width: 100%;
-	}
+	}
```
These line endings were still in LF

Pyther99 commented on VocaDbWeb/wwwroot/Content/Styles/base.less:
```diff
@@ -166,8 +166,8 @@ body .container-fluid {
 
-.vdb .page-title {
-	margin-bottom: 10px;
+.vdb .page-title {
+	margin-bottom: 10px;
```
These were also in LF

Pyther99 commented on VocaDbWeb/wwwroot/Content/Styles/base.less:
```diff
@@ -345,6 +345,12 @@ body .container-fluid {
 	}
 }
 
+.vcenter {
+	min-height: 100%;
+	min-height: 100vh;
+	display: flex;
+	align-items: center;
+}
```
Could be removed by upgrading bootstrap

Pyther99 commented on VocaDbWeb/Views/Song/Create.cshtml:
```diff
@@ -16,156 +16,176 @@
-					<tbody>
 						<tr>
+							<td class="formLabel">
+								<a style="text-decoration: underline dashed">
+									@Html.LabelFor(m => m.NameRomaji, htmlAttributes: new { onclick = "window.open('https://wiki.vocadb.net/SiteSettings/Wiki/Index/20?title=Romanization-walkthrough')" })
+								</a>
```
Probably needs a better implementation

Pyther99 commented:
Screenshot:
![image](https://user-images.githubusercontent.com/65015656/113159187-b195a700-923c-11eb-9b5e-ee4e6c222152.png)

Pyther99 commented:
![localhost_44398_Song_Create (1)](https://user-images.githubusercontent.com/65015656/113159826-34b6fd00-923d-11eb-9c26-2c5702f6bb22.png)

Pyther99 commented:
![localhost_44398_Song_Create](https://user-images.githubusercontent.com/65015656/113190631-d13cc780-925c-11eb-8ff4-068c14a3dfb0.png)

Pyther99 commented on VocaDbWeb/Views/Shared/KnockoutPartials/_CoverArtistMessage.cshtml:
```diff
@@ -0,0 +1,16 @@
+-->
+<div class="alert" style="display:none" data-bind="visible: coverArtists() && coverArtists().length > 0 && !canHaveOriginalVersion()">
+	<partial name="Partials/Shared/_NotificationIcon" />
+	@ViewRes.EntryCreateStrings.FoundCoverArtist
```
Causes the AppVeyor build to fail. Will be resolved after the changes from #817 get merged into this PR.

ycanardeau commented:
@Pyther99 Deployed on the beta site.

dset0x commented:
@Pyther99 Fantastic work turning the mockup into a functioning page already! It's so cool to see this on the beta site.

I wrote some feedback in this screenshot,
![song_submit_feedback](https://user-images.githubusercontent.com/8993055/114074634-dfb26100-98a4-11eb-8d75-516d7fa4670f.png)

Pyther99 commented:
@dset0x  The markup parsing error occurred because the description was missing a space between ### and the text.

I have fixed the side warnings to the vertical center of the viewport, but this can cause the warnings to overlap if there are enough duplicates. (This worsened the responsiveness for mobile devices and needs to be fixed in the future, see #6)

I also implemented a Bootstrap tooltip for the artist warning.

dset0x commented:
@Pyther99 That's awesome, thanks for tending to these.

If it worked better on mobile without the fixed location, a workaround could be to only enable that using `@media`, but I can't say how bad that might be.

Yet another minor thing. Not everyone is familiar with the initials "PV". "Promotional Video" might also be misleading though.

hikaru-soeda commented:
@Pyther99 I think the width of the label in the title input field should be increased.
It doesn't look good in the Japanese UI because the labels are vertical.
![Web キャプチャ_9-4-2021_195452_beta vocadb net_](https://user-images.githubusercontent.com/76765993/114170346-c88b7600-996d-11eb-9260-ca31016980b5.jpeg)

dset0x commented:
Looking at the current beta site, I it feels a bit awkwardly long now,
![image](https://user-images.githubusercontent.com/8993055/114263134-1f339700-99e4-11eb-96ed-de7eb3b34335.png)
But if it needs to be so that translations fit better, then it's no issue

Pyther99 commented:
@dset0x The width was reduced in the last commit, which is not yet deployed on the beta site. Nevertheless, I still shortened the width a little.
![localhost_44398_Song_Create (3)](https://user-images.githubusercontent.com/65015656/114273548-c2070800-9a1a-11eb-8f5c-e8205f8067dc.png)

ycanardeau commented:
Thanks for the PR!
